From 9d8a911ccfd36652d34d647abd98c3e7ba9dc157 Mon Sep 17 00:00:00 2001
From: kszaq <737984+kszaq@users.noreply.github.com>
Date: Thu, 29 Dec 2022 20:19:58 +0100
Subject: [PATCH] AMLCodec: always try to dequeue buffer in GetPicture

Return VC_BUFFER only if buffer is not empty, codec can't
dequeue data (get new picture) and buffer is not full.
---
 xbmc/cores/VideoPlayer/DVDCodecs/Video/AMLCodec.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/xbmc/cores/VideoPlayer/DVDCodecs/Video/AMLCodec.cpp b/xbmc/cores/VideoPlayer/DVDCodecs/Video/AMLCodec.cpp
index ffa268eb33..e91934661e 100644
--- a/xbmc/cores/VideoPlayer/DVDCodecs/Video/AMLCodec.cpp
+++ b/xbmc/cores/VideoPlayer/DVDCodecs/Video/AMLCodec.cpp
@@ -2532,7 +2532,7 @@ CDVDVideoCodec::VCReturn CAMLCodec::GetPicture(VideoPicture *pVideoPicture)
     return CDVDVideoCodec::VC_ERROR;
 
   float timesize(GetTimeSize());
-  if(!m_drain && timesize < 0.2f)
+  if(!m_drain && timesize == 0.0f)
     return CDVDVideoCodec::VC_BUFFER;
 
   if (DequeueBuffer() == 0)
